File: Scala Kompics/src/se/sics/kompics/scala/performancetest/kompics/PerfectPointToPointLink.scala
package se.sics.kompics.scala.performancetest.kompics

import se.sics.kompics.PortType
import se.sics.kompics.address.Address
import se.sics.kompics.Event

case class Pp2pSend(val dest: Address, val msg: Any) extends Event
case class Pp2pDeliver(val src: Address, val m: Any) extends Event

object PerfectPointToPointLink extends PortType {
	PortType.preloadInstance(this);
	request(classOf[Pp2pSend]);
	indication(classOf[Pp2pDeliver]);	
}

File: Java Kompics/src/se/kth/ii2202/experiments/common/PerfectPointToPointLink.java
package se.kth.ii2202.experiments.common;

import se.sics.kompics.PortType;

/**
 *
 * @author carbone
 */
public class PerfectPointToPointLink extends PortType {

    {
        request(Pp2pSend.class);
        indication(Pp2pDeliver.class);
    }

    
}
